int64 id
int64 cnt
int64 seq
float64 u0
float64 v0
float64 u1
float64 v1